packages/components/src/Avatar/Avatar.tsx

```diff
@@ -0,0 +1,104 @@
+import React, { CSSProperties } from "react";
+import classnames from "classnames";
+import Color from "color";
+import styles from "./Avatar.css";
+import { Icon } from "../Icon";
+
+interface AvatarProps {
+  /**
+   * A path to the image that will be displayed
+   */
+  readonly imageUrl?: string;
+  /**
+   * A users name to be used for assistive technology
+   */
+  readonly name?: string;
+  readonly initials?: string;
+  readonly color?: string;
+  /**
+   * @default 'medium'
+   */
+  readonly size?: "medium" | "large";
+}
+
+export function Avatar({
+  imageUrl,
+  name,
+  initials,
+  size = "medium",
+  color,
+}: AvatarProps) {
+  const style: CSSProperties = {
+    backgroundColor: color,
+    borderColor: color,
+  };
+
+  if (imageUrl) {
+    style.backgroundImage = `url(${imageUrl})`;
+  }
+
+  const className = classnames(styles.avatar, {
+    [styles.large]: size === "large",
+    [styles.hasBorder]: imageUrl && color,
+    [styles.isDark]: color && isDark(color),
+  });
+
+  return (
+    <div
+      className={className}
+      style={style}
+      role={imageUrl && "img"}
+      aria-label={name}
+    >
+      {initials && !imageUrl ? <Initials initials={initials} /> : ""}
+      {!initials && !imageUrl ? (
+        <Icon name="person" color={color && isDark(color) ? "white" : "blue"} />
```

Comment:
Color prop here might be clearer if we pre-compute it into an `iconColor` variable perhaps? The combination of the `&&` and the ternary I'm finding a bit hard to follow.